src/convertToObject.js

```diff
@@ -7,6 +7,30 @@
  */
 function convertToObject(sourceString) {
   // write your code here
+  const result = {};
+
+  const declarations = sourceString.split(';');
+
+  for (let declaration of declarations) {
+    declaration = declaration.trim();
+
+    if (declaration === '') {
+      continue;
+    }
+
+    const [property, value] = declaration.split(':');
+
+    if (!property || value === undefined) {
```

Comment:
The check `value === undefined` is appropriate here to ensure that both property and value are present. This prevents adding undefined values to the result object.